meteor/client/ui/TestTools/Timeline.tsx

```diff
@@ -183,20 +176,22 @@ function FilterInput({ filterChanged }: FilterInputProps) {
 
 interface TimelineStateTableProps {
 	now: number
-	allStates: ResolvedStates | undefined
+	resolvedTimeline: ResolvedTimeline | undefined
 }
-function TimelineStateTable({ allStates, now }: TimelineStateTableProps) {
+function TimelineStateTable({ resolvedTimeline, now }: TimelineStateTableProps) {
 	const [viewTime, setViewTime] = useState<number | null>(null)
 	const selectViewTime = useCallback((e: React.ChangeEvent<HTMLSelectElement>) => {
 		const val = Number(e.target.value)
 		setViewTime(isNaN(val) ? null : val)
 	}, [])
 
-	const state = allStates ? Resolver.getState(allStates, viewTime ?? now) : undefined
-	const times = _.uniq((allStates?.nextEvents ?? []).map((e) => e.time))
-
 	const [layerFilter, setLayerFilter] = useState<FilterInputValue>(undefined)
 
+	if (!resolvedTimeline) return null
+
+	const state = getResolvedState(resolvedTimeline, viewTime ?? now, 1)
```

Comment:
Is this cheap enough to do for every paint, or should this be inside a `useMemo` call? 
It used to be inside a `useMemo`, but that may have been excessive

Reply:
It should be fine, `getResolvedState` is functionally equivalent to the previous `Resolver.getState`, and is a fairly fast operation